### PR TITLE
fix: Update sleuth conda environment and add multithreading to kallisto quant

### PR DESCRIPTION
**Pull request by Austin-s-h**
While encountering a few errors related to differential expression, I upgraded the environment for running Sleuth and increased the efficiency of the kallisto quant rule.

Also, found a bug in the samples schema, batch_effect was missing.

### CODE REVIEW

File: workflow/schemas/samples.schema.yaml [hunk 1]
```diff
@@ -8,6 +8,9 @@ properties:
   condition:
     type: string
     description: sample condition that will be compared during differential expression analysis (e.g. a treatment, a tissue time, a disease)
-
+  batch_effect:
+    type: string
+    description: The batch that the sample belongs to
+
```
Comment (dlaehnemann) [inline workflow/schemas/samples.schema.yaml:14]: As the column name used in the `config["diffexp"]["models"]["modelX"]` specifications of `full` and `reduced` models is arbitrary, so I'm not sure if we need to include this in the schema (even if we don't include among the `required:` fields listed below). Does specifying this here give you a clear benefit (like a more useful error message in some case) or was this just meant for consistency?

File: workflow/rules/quant.smk [hunk 2]
```diff
@@ -23,6 +23,11 @@ rule kallisto_quant:
         extra=kallisto_params,
     conda:
         "../envs/kallisto.yaml"
+    # around 4 gb of memory usage with 1 thread. (hg38)
+    # over 8gb peak memory usage with 8 threads. (hg38)
```
Comment (dlaehnemann) [inline workflow/rules/quant.smk:27]: Thanks for checking this out and documenting the info!

File: workflow/rules/quant.smk [hunk 3]
```diff
@@ -23,6 +23,11 @@ rule kallisto_quant:
...
+    threads: 8
+    resources:
+        mem_mb=10000,
     shell:
         "kallisto quant -i {input.idx} -o {output} "
-        "{params.extra} {input.fq} 2> {log}"
+        "{params.extra} -t {threads} {input.fq} 2> {log}"
```
Comment (dlaehnemann) [inline workflow/rules/quant.smk:33]: While we are at it, I'd suggest moving to using the respective wrapper. This makes maintenance easier in the long run:
https://snakemake-wrappers.readthedocs.io/en/stable/wrappers/kallisto/quant.html

Same for the `rule kallisto_index` above:
https://snakemake-wrappers.readthedocs.io/en/stable/wrappers/kallisto/index.html

**Review by dlaehnemann — CHANGES_REQUESTED** [review]
Thanks again for taking the initiative of improving the workflow! I've added some suggestions.

File: workflow/schemas/samples.schema.yaml (hunk 1 above)
Comment (Austin-s-h) [reply workflow/schemas/samples.schema.yaml:14]: It turned out not to be so arbitrary, I think that it should be enforced.

File: workflow/rules/quant.smk (hunk 3 above)
Comment (Austin-s-h) [reply workflow/rules/quant.smk]: Should be resolved! Please review @dlaehnemann

File: workflow/schemas/samples.schema.yaml (hunk 1 above)
Comment (dlaehnemann) [reply workflow/schemas/samples.schema.yaml:14]: Ah, but it should be. If you simply name such a column differently in the `samples.tsv` file, for example `day_of_processing`, you can simply adjust the formulas in `config.yaml` to:
```
full: ~condition + day_of_processing
reduced: ~day_of_processing
```

Or if you don't have any batch effects, you can also do something like:
```
full: ~condition
reduced: ~1
```

Is this explanation clearer?

**Pull request closed without merging**